### PR TITLE
Fantomas: use nuget version

### PR DESCRIPTION
**Pull request by DedSec256**

### CODE REVIEW

File: rider-fsharp/build.gradle.kts [hunk 1]
```diff
@@ -240,19 +241,26 @@ tasks {
             files = files + listOf("$testHostName.dll", "$testHostName.pdb")
         }
 
-        files.forEach {
-            from(it) { into("${intellij.pluginName}/dotnet") }
+        val filesMapping = files.zip(
+            files.map {
+                val pathEnd = if (it.contains("Fantomas")) "/fantomas" else ""
+                "${intellij.pluginName}/dotnet$pathEnd"
```
Comment (auduchinok) [inline rider-fsharp/build.gradle.kts:247]: Could you use separate lists or additional relative path params for this? The `contains("Fantomas")` check looks too ad-hoc.

File: ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs [hunk 2]
```diff
@@ -21,23 +25,33 @@ static FantomasAssemblyResolver()
...
       return null;
     }
+
+    public static Assembly Resolve(object sender, ResolveEventArgs eventArgs)
+    {
+      var assemblyName = $"{new AssemblyName(eventArgs.Name).Name}.dll";
+      var paths =
+        assemblyName == "Fantomas.dll" || assemblyName == "FSharp.Compiler.Service.dll"
+          ? (IEnumerable<string>) FSharpFantomasAssembliesPath
+          : OurAdditionalProbingPaths;
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs:46]: Just a thought: what if you just prepend Fantomas-related paths, so it searches them first? Will it work?

Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs:46]: Do you want to add it to the top of the OurAdditionalProbingPaths list? If Fantomas' FCS assembly is suddenly not found, the search will continue in OurAdditionalProbingPaths and the wrong version will be loaded (custom version from F# plugin).

Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs:46]: If it's suddenly not found we're already in the trouble anyway, no?

Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs:46]: > Do you want to add it to the top of the OurAdditionalProbingPaths list?

Yup, so it'd effectively revert to previous logic, it seemed a bit less complicated. Not a big issue, though.

File: ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs [hunk 3]
```diff
@@ -8,14 +8,20 @@ namespace JetBrains.ReSharper.Plugins.FSharp.Fantomas.Host
...
     static FantomasAssemblyResolver()
     {
-      var paths = Environment.GetEnvironmentVariable(AdditionalProbingPathsEnvVar);
-      if (string.IsNullOrWhiteSpace(paths)) return;
+      var fantomasPath = Environment.GetEnvironmentVariable(FantomasAssembliesPathEnvVar);
+      var riderPaths = Environment.GetEnvironmentVariable(AdditionalProbingPathsEnvVar);
+      if (string.IsNullOrWhiteSpace(fantomasPath))
+        throw new ArgumentException("Argument IsNullOrWhiteSpace", FantomasAssembliesPathEnvVar);
+      if (string.IsNullOrWhiteSpace(riderPaths))
+        throw new ArgumentException("Argument IsNullOrWhiteSpace", AdditionalProbingPathsEnvVar);
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasAssemblyResolver.cs:21]: Use `Assertion.Assert`.

File: rider-fsharp/build.gradle.kts [hunk 4]
```diff
@@ -231,8 +231,9 @@ configure<RdGenExtension> {
 
 tasks {
     withType<PrepareSandboxTask> {
-        var files = libFiles + pluginFiles.map { "$it.dll" } + pluginFiles.map { "$it.pdb" } + typeProvidersFiles + fantomasFiles
+        var files = libFiles + pluginFiles.map { "$it.dll" } + pluginFiles.map { "$it.pdb" } + typeProvidersFiles
         files = files.map { "$resharperPluginPath/src/$it" }
+        val fantomasFiles = fantomasFiles.map { "$resharperPluginPath/src/$it" }
```
Comment (auduchinok) [inline rider-fsharp/build.gradle.kts:236]: Will `{ "../fantomas/$it" }` work?

Comment (DedSec256) [reply rider-fsharp/build.gradle.kts:236]: No, because Fantomas.Host is contained in ReSharper.FSharp

Comment (auduchinok) [reply rider-fsharp/build.gradle.kts:236]: Ah, that's the from path, my bad.
Thanks!

**Review by auduchinok — APPROVED** [review]

**Pull request merged**